src/glang/codegen/builtin_callables.py

```diff
@@ -698,6 +700,198 @@ def assert_can_write_to(self) -> None:
         self._array_ptr.assert_can_write_to()
 
 
+class TrapExpression(BuiltinCallable):
+    def __init__(
+        self,
+        specialization: list[SpecializationItem],
+        arguments: list[TypedExpression],
+        meta: Meta,
+    ) -> None:
+        assert len(specialization) == 0
+        assert len(arguments) == 0
+        StaticTypedExpression.__init__(
+            self,
+            VoidType(),
+            Type.Kind.VALUE,
+            meta,
+        )
+
+    def __repr__(self) -> str:
+        return "TrapExpression()"
+
+    def generate_ir(self, ctx: IRContext) -> IROutput:
+        ir = IROutput()
+        dbg = self.add_di_location(ctx, ir)
+        ir.lines.append(f"call void @llvm.trap(), {dbg}")
```

Comment:
What does this lower to on x86? From the reference
> This intrinsic is lowered to the target dependent trap instruction. If the target does not have a trap instruction, this intrinsic will be lowered to a call of the abort() function.

Noting that we don't have an `abort()` function

Reply:
https://godbolt.org/z/E14s5zK9v which gives a https://mudongliang.github.io/x86/html/file_module_x86_id_318.html. Which triggers a `SIGILL (Illegal instruction)` on linux

Looks like arm also has one: https://godbolt.org/z/